packages/components/shared/src/directives/anchor.ts

```diff
@@ -23,6 +23,15 @@ export class AnchorDirective extends Directive {
   #cleanup?: () => void;
   #config?: AnchorDirectiveConfig;
   #host?: HTMLElement;
+  visible: boolean = true;
+
+  observer = new IntersectionObserver(
+    (entries, _) => {
+      const outOfView = entries.some(e => e.intersectionRatio < 0.5);
+      updatePopoverVisibility(this.#host, outOfView);
+    },
+    { threshold: 0.5 }
```

Comment:
Maybe we can add possibility for developers to set `rootMargin` there? Or whole `options` object for the intersection observer? Then they would be able to fit it to their needs (like sticky header size)?  And then we can also show it in the story example with sticky header that it could hide below the sticky header.

Reply:
Oohhh, that's smart I'll look into that.